### ISSUE TITLE
Tribler fails to chew a mega-torrent

### ISSUE THREAD

**Issue by vi** [Tribler/tribler#3609]
##### Tribler version/branch+revision: #####

v7.0.2

##### Operating system and version: #####

Linux

##### Steps to reproduce the behavior: #####

1. Download [this torrent file](https://sites.google.com/site/tlmcfiles/Touhou%20lossless%20music%20collection%20v.19.torrent) (9.2M);
2. "Import torrent from file"
3. Choose the file
4. Deselect everything except of the first small-ish entry
5. Press Download

##### Expected behavior: #####

The torrent starts downloading and also becomes searchable globally, like usual.

##### Actual behavior: #####

Error appears.

##### Relevant log file output: #####

```
/home/tribler/tribler/Tribler/Core/Modules/restapi/downloads_endpoint.py:295:render_PUT
/home/tribler/tribler/Tribler/Core/Session.py:313:start_download_from_uri
/home/tribler/tribler/Tribler/Core/Libtorrent/LibtorrentMgr.py:571:start_download_from_uri
/home/tribler/tribler/Tribler/Core/Libtorrent/LibtorrentMgr.py:618:start_download
/home/tribler/tribler/Tribler/Core/TorrentDef.py:105:load_from_memory
/home/tribler/tribler/Tribler/Core/TorrentDef.py:119:_create
/home/tribler/tribler/Tribler/Core/Utilities/utilities.py:44:validTorrentFile
]
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/twisted/web/server.py", line 190, in process
    self.render(resrc)
  File "/usr/lib/python2.7/dist-packages/twisted/web/server.py", line 241, in render
    body = resrc.render(self)
  File "/usr/lib/python2.7/dist-packages/twisted/web/resource.py", line 250, in render
    return m(request)
  File "/home/tribler/tribler/Tribler/Core/Modules/restapi/downloads_endpoint.py", line 295, in render_PUT
    unquote_plus(unicode(parameters['uri'][0], 'utf-8')), download_config)
  File "/home/tribler/tribler/Tribler/Core/Session.py", line 313, in start_download_from_uri
    return self.lm.ltmgr.start_download_from_uri(uri, dconfig=dconfig)
  File "/home/tribler/tribler/Tribler/Core/Libtorrent/LibtorrentMgr.py", line 571, in start_download_from_uri
    return succeed(self.start_download(torrentfilename=argument, dconfig=dconfig))
  File "/home/tribler/tribler/Tribler/Core/Libtorrent/LibtorrentMgr.py", line 618, in start_download
    tdef = TorrentDef.load_from_memory(torrent_data)
  File "/home/tribler/tribler/Tribler/Core/TorrentDef.py", line 105, in load_from_memory
    return TorrentDef._create(data)
  File "/home/tribler/tribler/Tribler/Core/TorrentDef.py", line 119, in _create
    validTorrentFile(metainfo)
  File "/home/tribler/tribler/Tribler/Core/Utilities/utilities.py", line 44, in validTorrentFile
    raise ValueError('announce URL bad')
ValueError: announce URL bad
```

**Comment by xoriole** [comment]
@vi Thank you for the reporting. I can reproduce the issue so will fix it soon.

**Comment by xoriole** [comment]
It seems an empty announce URL was causing this issue for which the fix was straight forward.

However, adding this torrent to Tribler highlighted a more serious issue. The startup time increased significantly. Note that the total size of the files included in this torrent is **1.7TB** but even on selecting a single small file to download, the startup time is still significantly large. 

On a positive side, this torrent seems like a good testcase for debugging and fixing this issue.

**Comment by vi** [comment]
Note: previous issue about similar file: #1689.
